### PR TITLE
chore: Change user search name surname highlight logic #WPB-6536

### PR DESCRIPTION
**Pull request by m-zagorski**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6536" title="WPB-6536" target="_blank">BugWPB-6536</a>  [Android] Highlight missing at @ sign while searching for a user
  </td></table>
  <br />
 

https://wearezeta.atlassian.net/browse/WPB-6536

# What's new in this PR?

### Issues

When we're searching for users, we were not highlighting `@` 

### Solutions

We take into considerations the prefix, and now queries with `@` highlight properly. There is an edge case where our search for `@mat` would return results without `@` and we do not highlight anything, but I confirmed with Vina that it is fine in that case.

#### How to Test

Search for users with `@` and without, check if proper parts of name and surname are highlighted. In the task description there are also places where it shouldnt change after this changes

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by borichellow — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/HighLightName.kt [hunk 1]
```diff
@@ -40,14 +40,12 @@ fun HighlightName(
     modifier: Modifier = Modifier
 ) {
 
-    val queryWithoutSuffix = searchQuery.removeQueryPrefix()
-
     val highlightIndexes = QueryMatchExtractor.extractQueryMatchIndexes(
-        matchText = queryWithoutSuffix,
+        matchText = searchQuery,
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/search/HighLightName.kt:44]: when the user add @ it means to do search based on user handle, adding it to the search query will result to handle search being done with @ in the beginning 
for example my is a handle is "mo_wirer" then this is what will be stored in the database and I need to do handle search then the user enter "@mo_wire" and the actual search in the database will be against "@mo_wire" which will result to no valid results

Comment (m-zagorski) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/search/HighLightName.kt:44]: Totally agree, but the logic for search itself didnt change, we're still doing it as you explained, so if `@` is present it will be done like that, as it was before, this changes only the way its highlighted, so if we're looking for `@mat` and got results like:
- mat mat - wont highlight anything
- @mateusz.123 - will highlight @mat

The same goes if the query was done without `@` so if we search for `mat` it would be:
- mat mat - both maths are going to be highlighted
-@mateusz.123 - `mat` will get highlighted

Those changes will not affect any search results in any case

**Review by saleniuk — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/HighLightName.kt (hunk 1 above)
Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/search/HighLightName.kt:44]: ok got it
